braindecode/models/eegresnet.py

```diff
@@ -31,14 +33,24 @@ def __init__(self, in_chans,
                  batch_norm_epsilon=1e-4,
                  conv_weight_init_fn=lambda w: init.kaiming_normal_(w, a=0)):
         super().__init__()
+        self.in_chans = in_chans
+        self.n_classes = n_classes
+        self.input_time_length = input_time_length
         if final_pool_length == 'auto':
             assert input_time_length is not None
         assert first_filter_length % 2 == 1
-        self.__dict__.update(locals())
-        del self.self
+        self.final_pool_length = final_pool_length
+        self.n_first_filters = n_first_filters
+        self.n_layers_per_block = n_layers_per_block
+        self.first_filter_length = first_filter_length
+        self.nonlinearity = nonlinearity
+        self.split_first_layer = split_first_layer
+        self.batch_norm_alpha = batch_norm_alpha
+        self.batch_norm_epsilon = batch_norm_epsilon
+        self.conv_weight_init_fn = conv_weight_init_fn
```

Comment:
Makes me cry but ok ;)

Reply:
using locals() is really brutal and it could pull some crap in the self.__dict__
one option is to inspect the __init__ and set all parameters passed that we could detect with inspect module.

something like:

```
import inspect
init_signature = inspect.signature(self.__init__)
```

Reply:
Yeah that maybe possible like as separate function. But maybe outside this PR then.
What is your view on data classes? https://docs.python.org/3/library/dataclasses.html

Reply:
Btw some small discussion on this also here: https://stackoverflow.com/a/12191200/1469195

Reply:
Created new issue for this https://github.com/braindecode/braindecode/issues/93